### DISCUSSION TITLE
Is there a way to word wrap based on a certain character or regex?

### DISCUSSION THREAD

**Discussion by Jbleezy** [microsoft/vscode-discussions#1042] [Extension Development QnA] [answered]
For a project I am working on, I have certain files that are mostly one very long line of text (usually around 20,000 characters) that has keys and values with a `\` as the delimiter.

It would be very nice to be able to have word wrap happen after every 2nd `\`, but I can't think of a way to do this. The closest I can get would be to make some keyboard shortcut that adds actual newlines, but I would have to constantly keep converting back and forth because keeping those newlines would make the file type not work.

**Comment by alefragnani** [comment, marked as answer]
Hi @Jbleezy ,

There is no such setting in VS Code. The current word wrap approach is based on the number of characters or viewport size. And doing a quick look in the marketplace, I couldn't find any extension that fit your needs.

If your request is to VS Code provide such feature, I suggest you to open an issue in VS Code repo (https://github.com/microsoft/vscode/issues/new?assignees=&labels=&projects=&template=feature_request.md) providing the details. If the team understand this would be better fitted to an extension, they would label the issue accordingly and someone in the community could contribute.

On the other hand, if you intend to develop an extension yourself, I see two possible alternatives to try out:

### Custom Editor API

Using the Custom Editor API (https://code.visualstudio.com/api/extension-guides/custom-editors) you could would be able to render those long lines with your own word wrap strategy without necessarily modifying its content format. It may be the more complex solution, but should also be the most powerful one.

### Commands and Hooks

Leave the long line as it is, with its 20k chars, untouched, and build your extension to work with it. Create commands to wrap and unwrap the lines based on its delimiters, so the user can prepare the file for easier editing and revert back to the original format. If the unwrap approach is not a _good_ alternative, maybe you could try out monitoring the _save_ event, using `onWillSaveTextDocument` and `onDidSaveTextDocument` events.

Hope this helps
